sweep/sweeper.go

```diff
@@ -680,7 +681,80 @@ func (s *UtxoSweeper) removeExclusiveGroup(group uint64) {
 		s.signalAndRemove(&outpoint, Result{
 			Err: ErrExclusiveGroupSpend,
 		})
+
+		// Make sure we remove conflicting transactions from the wallet.
+		err := s.removeConflictingSweepTx(outpoint)
+		if err != nil {
+			log.Warnf("Unable to remove conflicting sweep tx from "+
+				"wallet for outpoint %v : %v", outpoint, err)
+		}
+	}
+}
+
+// removeConflictingSweepTx removes any unconfirmed sweep transaction from the
+// wallet which spends the passed outpoint. It is necessary to cleanup the
+// unmined tx store for wallets which use neutrino as a backend.
+// Most common case is when a channel is force closed and anchor cpfp txns are
+// created to bump the initial commitment transaction. In this case an achor
+// cpfp is broadcasted for up to 3 commitment transactions (local,
+// remote-dangling, remote). Using neutrino all of those transactions will be
+// accepted (the commitment tx will be different in all of those cases) and have
+// to be removed as soon as one of them confirmes (they do have the same
+// ExclusiveGroup).
+func (s *UtxoSweeper) removeConflictingSweepTx(outpoint wire.OutPoint) error {
```

Comment:
if we pass a set of outpoints(`map[wire.OutPoint]struct{}`) then this method is identical to `removeLastSweepDescendants`?

Reply:
very good idea

Reply:
renamed to the following signature:

`func (s *UtxoSweeper) removeConflictSweepDescendants(outpoints map[wire.OutPoint]struct{}) error {`